R/assertions.R

```diff
@@ -1714,27 +1716,29 @@ assert_one_to_one <- function(dataset, vars1, vars2) {
 #'   dataset = my_data,
 #'   var = USUBJID
 #' ))
-assert_date_var <- function(dataset, var, dataset_name = NULL, var_name = NULL) {
+assert_date_var <- function(dataset,
+                            var,
+                            dataset_name = rlang::caller_arg(dataset),
+                            var_name = rlang::caller_arg(var),
+                            message = NULL,
+                            class = "assert_date_var",
+                            call = parent.frame()) {
   var <- assert_symbol(enexpr(var))
   assert_data_frame(dataset, required_vars = exprs(!!var))
-  assert_character_scalar(dataset_name, optional = TRUE)
-  assert_character_scalar(var_name, optional = TRUE)
+  assert_character_scalar(dataset_name)
+  assert_character_scalar(var_name)
   column <- pull(dataset, !!var)
-  if (is.null(dataset_name)) {
-    dataset_name <- arg_name(substitute(dataset))
-  }
-  if (is.null(var_name)) {
-    var_name <- as_label(var)
-  }
+
   if (!is.instant(column)) {
-    abort(paste0(
-      "`",
-      var_name,
-      "` in dataset `",
-      dataset_name,
-      "` is not a date or datetime variable but is ",
-      friendly_type_of(column)
-    ))
+    message <- message %||%
+      "Column {.val {var_name}} in data frame {.code {dataset_name}} must be
```

Comment:
```suggestion
      "Column {.val {var_name}} in dataset {.code {dataset_name}} must be
```
To be consistent with documentation and other messages?